Firmware/Marlin_main.cpp

```diff
@@ -2820,15 +2820,15 @@ static void gcode_G80()
     run = false;
 #endif //PINDA_THERMISTOR
 
-    uint8_t nMeasPoints = code_seen('N') ? code_value_uint8() : eeprom_read_byte((uint8_t*)EEPROM_MBL_POINTS_NR);
-    if (nMeasPoints != 7) {
-        nMeasPoints = 3;
-    }
+    uint8_t nMeasPoints = eeprom_read_byte((uint8_t*)EEPROM_MBL_POINTS_NR);
+    bool bCodeSeen = code_seen('N');
```

Comment:
Hmm, don't really like that this variable is shared between the two arguments parsing. Could the code_seen be used directly in the if statement and avoid declaring bCodeSeen in the first place?

Reply:
If it's used in the if statement, it's has to be executed after the value declaration. (You can only do one declaration inside the if statement and it has to be the first thing you do). That won't work because code_value_uint8 has to be called after code_seen so the pointer is set beforehand.

I would have preferred to to:
```
bool bCodeSeen;

uint8_t nMeasPoints = eeprom_read_byte((uint8_t*)EEPROM_MBL_POINTS_NR);
bCodeSeen = code_seen('N');
...
```
To make it look proper, but somehow that leads to higher mem usage too... 🤷 😄

Reply:
I could do:
```
uint8_t nMeasPoints = eeprom_read_byte((uint8_t*)EEPROM_MBL_POINTS_NR);
    if (uint8_t bCodeSeen = code_seen('N'), value = code_value_uint8(); bCodeSeen && (value == 7 || value == 3))
      nMeasPoints = value;
```
Also not super sexy, but same mem usage.